From 1e910ef672f102978f9add5da6ab0b10ae8cd9c5 Mon Sep 17 00:00:00 2001
From: Zahari Doychev <zdoychev@maxlinear.com>
Date: Mon, 4 Aug 2025 11:31:09 +0200
Subject: [PATCH 3/4] dsa: mxl862xx: Remove BR_PORT_LOCKED flag

The BR_PORT_LOCKED flag is not present in the Linux 5.15 kernel.
Remove this flag to enable successful compilation with the PRX 5.15
kernel.
---
 linux/drivers/net/dsa/mxl862xx/mxl862xx.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/linux/drivers/net/dsa/mxl862xx/mxl862xx.c b/linux/drivers/net/dsa/mxl862xx/mxl862xx.c
index 808e99a..83b08e1 100644
--- a/linux/drivers/net/dsa/mxl862xx/mxl862xx.c
+++ b/linux/drivers/net/dsa/mxl862xx/mxl862xx.c
@@ -4475,10 +4475,10 @@ static int mxl862xx_port_pre_bridge_flags(struct dsa_switch *ds, int port,
 {
 	int ret = 0;
 
-	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
+	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
 		dev_err(ds->dev, "%s: Port:%d unsupported bridge flags:0x%lx\n",
 				__func__, port, flags.mask);
-		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
+		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
 			ret = -EINVAL;
 			NL_SET_ERR_MSG_MOD(extack, "Unsupported bridge flags:0x%lx");
 		}
@@ -4501,10 +4501,10 @@ static int mxl862xx_port_bridge_flags(struct dsa_switch *ds, int port,
 
 	/* .port_pre_bridge_flags is called after this function,
 	 *  so the supported flags check is needed also here */
-	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
+	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
 		dev_err(ds->dev, "%s: Port:%d unsupported bridge flags:0x%lx\n",
 				__func__, port, flags.mask);
-		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
+		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
 			ret = -EINVAL;
 			goto EXIT;
 		}
-- 
2.43.0

